packages/cmake-rn/src/platforms/types.ts

```diff
@@ -47,9 +48,12 @@ export type Platform<
    */
   triplets: Readonly<Triplets>;
   /**
-   * Get the limited subset of triplets that should be built by default for this platform, to support a development workflow.
+   * Get the limited subset of triplets that should be built by default for this platform.
+   *
    */
-  defaultTriplets(): Triplets[number][] | Promise<Triplets[number][]>;
+  defaultTriplets(
+    purpose: "development" | "release",
```

Comment:
Xcode uses the terminology [build configuration](https://developer.apple.com/documentation/xcode/adding-a-build-configuration-file-to-your-project) (which you can name `"orange" | "banana"` if you wish, but defaults to `Debug` and `Release` when creating new projects).

They confusingly use the term `Development` for codesigning for local deployment.

Reply:
Oh right. Then surely this should be changed to `onlyBuildActiveArchitecture: boolean` or something? As "active architecture" I believe is terminology commonly used across various IDEs and build systems.

Or `buildAllArchitectures: boolean`.

Reply:
I agree - it's the same concept as the `activeArchOnly` option passable to the React Native CLI: [src/commands/buildAndroid/index.ts#L58-L77](https://github.com/react-native-community/cli/blob/dc848fd90842beb68e92f6ab7c9a00458a151b03/packages/cli-platform-android/src/commands/buildAndroid/index.ts#L58-L77) where it's actually probing ADB. We might be able to do the same or something similar to figure out what simulators and emulators are actively booted 🤔

That being said, I don't know if it's worth it though investing too much into this feature: We're running all of the builds of triplets in parallel so the increase in build-time doesn't scale linearly with the amount of triplets.